src/platform/macos.rs

```diff
@@ -113,6 +114,8 @@ pub trait WindowBuilderExtMacOS {
     fn with_has_shadow(self, has_shadow: bool) -> WindowBuilder;
     /// Window accepts click-through mouse events.
     fn with_accepts_first_mouse(self, accepts_first_mouse: bool) -> WindowBuilder;
+    /// Sets a parent to the window to be created.
+    fn with_parent_window(self, parent: *mut c_void) -> WindowBuilder;
```

Comment:
This is an `unsafe` operation (since we're taking a raw pointer), and it should be marked as such. Would also be nice with a `# Safety` section detailing exactly which guarantees the user must uphold

Reply:
No, I think it's correct that we take a `*mut c_void` pointer (unless your use-case allows it to just be `&Window`, see https://github.com/rust-windowing/winit/pull/2548#pullrequestreview-1189662991).

Reply:
I see. My use-cases needs to cover all NSWindow from others rather than just winit's window I think.
I'll make this `unsafe` and add `#Safety` doc.